src/vm/jitinterface.cpp

```diff
@@ -8522,7 +8522,7 @@ CEEInfo::getMethodSigInternal(
         // JIT should not generate shared generics aware call code and insert the secret argument again at the callsite.
         // Otherwise we would end up with two secret generic dictionary arguments (since the stub also provides one).
         //
-        BOOL isDefaultInterfaceMethodCallSite = isCallSite && ftn->IsDefaultInterfaceMethod();
+        BOOL isDefaultInterfaceMethodCallSite = isCallSite && (ftn->GetMethodTable()->IsInterface() && !ftn->IsStatic() && !ftn->IsAbstract());
```

Comment:
I meditated on this a little and I think the problem is in this line. The original condition was bogus and the new one doesn't fix it.

What is happening here is this:
* Calls to shared instance methods normally need an instantiation argument.
* Interface calls don't need an instantiation argument. We don't really know what specific type the dispatch will end up on. It could be a default interface method implementation, but it could also be a regular interface method implemented by a class (in which case we get the instantiation argument from `this`).
* Default interface methods feature put us into a situation where we need to report the instantiation argument *sometimes* though, because we could be actually doing a direct call to an interface method (not an "interface call"; but a call to a method whose address was known at compile time that happens to be owned by an interface type)

I think the specific condition for when we want to report this in `getMethodSigInternal` is when:

* The method got called from `getCallInfo` (the existing `isCallSite` parameter covers that)
* `ftn->RequiresInstArg()` is true (so no change there)
* The method is an instance method on an interface
* We are doing a direct call (`directCall` is `true` in `getCallInfo`)

Just leave this line as it was for now and the problem will probably disappear.

Reply:
Thanks for the explanation and looking into this! Changed back.